tools/clang/unittests/SPIRV/StructureTest.cpp

```diff
@@ -0,0 +1,184 @@
+  appendVector(&expected,
+               constructInst(spv::Op::OpCapability,
+                             {static_cast<uint32_t>(spv::Capability::Shader)}));
+
+  m.addExtension("ext");
+  const uint32_t extWord = 'e' | ('x' << 8) | ('t' << 16);
+  appendVector(&expected, constructInst(spv::Op::OpExtension, {extWord}));
+
+  m.addExtInstSet(5, "gl");
+  const uint32_t glWord = 'g' | ('l' << 8);
+  appendVector(&expected, constructInst(spv::Op::OpExtInstImport, {5, glWord}));
+
+  m.setAddressingModel(spv::AddressingModel::Logical);
+  m.setMemoryModel(spv::MemoryModel::GLSL450);
+  appendVector(
+      &expected,
+      constructInst(spv::Op::OpMemoryModel,
+                    {static_cast<uint32_t>(spv::AddressingModel::Logical),
+                     static_cast<uint32_t>(spv::MemoryModel::GLSL450)}));
+
+  m.addEntryPoint(spv::ExecutionModel::Fragment, 2, "main", {42});
+  const uint32_t mainWord = 'm' | ('a' << 8) | ('i' << 16) | ('n' << 24);
+  appendVector(
+      &expected,
+      constructInst(spv::Op::OpEntryPoint,
+                    {static_cast<uint32_t>(spv::ExecutionModel::Fragment), 2,
+                     mainWord, /* addtional null in name */ 0, 42}));
+
+  m.addExecutionMode(constructInst(
+      spv::Op::OpExecutionMode,
+      {2, static_cast<uint32_t>(spv::ExecutionMode::OriginUpperLeft)}));
+  appendVector(&expected,
+               constructInst(spv::Op::OpExecutionMode,
+                             {2, static_cast<uint32_t>(
+                                     spv::ExecutionMode::OriginUpperLeft)}));
+
+  // TODO: other debug instructions
+
+  m.addDebugName(2, llvm::None, "main");
+  appendVector(&expected,
+               constructInst(spv::Op::OpName,
+                             {2, mainWord, /* additional null in name */ 0}));
+
+  m.addDecoration(constructInst(
+      spv::Op::OpDecorate,
+      {2, static_cast<uint32_t>(spv::Decoration::RelaxedPrecision)}));
+  appendVector(&expected,
+               constructInst(spv::Op::OpDecorate,
+                             {2, static_cast<uint32_t>(
+                                     spv::Decoration::RelaxedPrecision)}));
+
+  m.addType(constructInst(spv::Op::OpTypeVoid, {1}));
+  appendVector(&expected, constructInst(spv::Op::OpTypeVoid, {1}));
+
+  m.addType(constructInst(spv::Op::OpTypeFunction, {3, 1, 1}));
+  appendVector(&expected, constructInst(spv::Op::OpTypeFunction, {3, 1, 1}));
+
+  // TODO: constant
+  // TODO: variable
+
+  auto f = Function(1, 2, spv::FunctionControlMask::MaskNone, 3);
+  auto bb = BasicBlock(4);
+  bb.addInstruction(constructInst(spv::Op::OpReturn, {}));
+  f.addBasicBlock(std::move(bb));
+  m.addFunction(std::move(f));
+  appendVector(&expected, constructInst(spv::Op::OpFunction, {1, 2, 0, 3}));
+  appendVector(&expected, constructInst(spv::Op::OpLabel, {4}));
+  appendVector(&expected, constructInst(spv::Op::OpReturn, {}));
+  appendVector(&expected, constructInst(spv::Op::OpFunctionEnd, {}));
+
+  m.setBound(6);
```

Comment:
why is the bound `6` and not `5`? it looks like the last id that is used is 4, so shouldn't the bound be 5?